File: Codevita-Problems-Solutions/mockvita-9-Phase-I/c.cpp
#include<bits/stdc++.h>
using namespace std ;
#define fo(i, n) for(int i = 0; i < n; i++)
#define foo(i, k, n) for(int i = k; i < n; i++)
#define deb(x) {cout << #x << " " << x << endl ;}
#define IOS ios::sync_with_stdio(false), cin.tie(0) ;
#define MOD = 1e9+7 ;
const int N = 1e6+7 ;

int main(int argc, char const *argv[]){
    IOS ;
    
    unordered_map<int, int> cache ;
    int cars ;
    cin >> cars ;
    int answer = 0 ;
    fo(i, cars) {
        int xaxis, yaxis, velocity ;
        cin >> xaxis >> yaxis >> velocity ;
        int tmp = sqrt(pow((xaxis/velocity), 2) + pow((yaxis/velocity),2)) ;
        cache[tmp]++ ;
    }
    for(auto &&pp : cache) {
        if(pp.second != 1) {
            answer += (pp.second * (pp.second - 1)) / 2;
        }
    }
    cout << answer ;

    return 0;
}

File: Codevita-Problems-Solutions/mockvita-9-Phase-I/c.py
import math

cache = {}
cars = int(input())
answer = 0

for i in range(cars):
    x,y,v = list(map(int, input().split()))
    # tmp = math.sqrt( (x/v)**2 + (y/v)**2 )
    tmp = math.sqrt( (x**2 + y**2) )/v
    if cache.get(tmp) == None:
        cache[tmp] = 1
    else:
        cache[tmp] += 1


for k in cache:
    if cache[k]!=1:
        answer += (cache[k] * (cache[k] - 1))/2


print(int(answer)) ;